cohorts/survival.py

```diff
@@ -72,59 +77,89 @@ def plot_kmf(df,
         if is_median:
             label_suffix += " (median)"
         default_label_with_condition = "%s > %s" % (condition_col, label_suffix)
-    else:
+        with_condition_label = with_condition_label or default_label_with_condition
+        no_condition_label = no_condition_label or default_label_no_condition
+        label_map = {False: no_condition_label,
+                     True: with_condition_label}
+        color_map = {False: no_condition_color,
+                     True: with_condition_color}
+    elif df[condition_col].dtype == 'O' or df[condition_col].dtype.name == "category":
+        condition = df[condition_col].astype("category")
+        label_map = dict()
+        [label_map.update({condition_value: '{} = {}'.format(condition_col,
+                                                    condition_value)})
+                 for condition_value in condition.unique()]
+        rgb_values = sb.color_palette("Set2", len(label_map.keys()))
```

Comment:
This is a good point - thanks. I didn't give this much thought because I was planning to allow the user to pick a "palette". Here I ended up hard-coding one, for sake of time, but I think I can allow the user to pass one in.